centaur/src/main/scala/centaur/test/metadata/WorkflowFlatMetadata.scala

```diff
@@ -79,17 +81,62 @@ object WorkflowFlatMetadata {
     }
   }
 
-  def fromMetadataJson(json: WorkflowMetadata): ErrorOr[WorkflowFlatMetadata] = {
-    import DefaultJsonProtocol._
-    Try(json.value.parseJson.asJsObject.flatten().convertTo[Map[String, JsValue]]) match {
-      case Success(m) => Valid(WorkflowFlatMetadata(m))
-      case Failure(e) => invalidNel(s"Unable to create Metadata from JSON: ${e.getMessage}")
+  def fromWorkflowMetadata(workflowMetadata: WorkflowMetadata): ErrorOr[WorkflowFlatMetadata] = {
+    val jsValue: ErrorOr[JsValue] = Try(workflowMetadata.value.parseJson) match {
+      case Success(jv) => Valid(jv)
+      case Failure(e) => invalidNel(s"Unable to create JsValue from String: ${e.getMessage}")
+    }
+
+    for {
+      jv <- jsValue
+      map <- jv.asMap
+    } yield WorkflowFlatMetadata(map)
+  }
+
+  implicit class EnhancedWorkflowMetadata(val workflowMetadata: WorkflowMetadata) {
+    def asFlat: WorkflowFlatMetadata = {
+      WorkflowFlatMetadata.fromWorkflowMetadata(workflowMetadata).unsafe
```

Comment:
TOL: Is there much value-add of `asFlat` over `fromWorkflowMetadata` (other than throwing away the `ErrorOr` wrapper)?

Reply:
that and the beauteous brevity that results 😛